agimus_controller/agimus_controller/trajectory.py

```diff
@@ -37,14 +140,53 @@ class WeightedTrajectoryPoint:
     point: TrajectoryPoint
     weight: TrajectoryPointWeights
 
+    def __eq__(self, other):
+        if not isinstance(other, WeightedTrajectoryPoint):
+            return False
+
+        # Compare the 'point' and 'weight' attributes
+        if self.point != other.point:
+            return False
 
-class TrajectoryBuffer(deque):
+        if self.weight != other.weight:
+            return False
+
+        return True
+
+
+class TrajectoryBuffer(list):
```

Comment:
Is there a particular reason why we inherit from list and not use a list internally?
It does not feel like a good idea to inherit from list to me.

Reply:
ok I will change it.
It's just that we have the append/pop method for free but I can add them